src/actors/symcaches.rs

```diff
@@ -222,35 +230,38 @@ impl SymCacheActor {
         let identifier = request.identifier.clone();
         let scope = request.scope.clone();
 
-        object.and_then(move |object| {
-            object
-                .map(move |object| {
-                    Either::A(symcaches.compute_memoized(FetchSymCacheInternal {
-                        request,
-                        objects_actor: objects,
-                        object_meta: object,
-                        threadpool,
+        find_result_future.and_then(move |find_result: FoundObject| {
+            let FoundObject { meta, candidates } = find_result;
+            let candidates: Arc<[ObjectCandidate]> = Arc::from(candidates);
+            meta.map(clone!(candidates, |object_meta| {
```

Comment:
hm, is the whole reason `candidates` is wrapped in an `Arc` this `clone!` here? I wonder if its worth converting to `async fn` to get rid of it?

Reply:
Kind off, but it doesn't really matter much.  This changes again in (#316) where we get a newtype for this is fine for now.